coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/ProposalBudgetNumberOfMonthsServiceImpl.java

```diff
@@ -16,118 +16,40 @@
-        double totalPartialMonthCount = 0;
-        int yearEnd = endDate.getYear();
-        boolean isLeapYear = ((yearEnd % 4 == 0) && (yearEnd % 100 != 0) || (yearEnd % 400 == 0));
-        
-        Date startDateAdded = rollDays(startDate, 1);
-        int previousMonth = startDateAdded.getMonth();
-        int previousMonthDayCount = 0;
-        while (startDateAdded.before(endDate)) {
-            startDateAdded = rollDays(startDateAdded, 1);
-            int currentMonth = startDateAdded.getMonth();
-            if (previousMonth != currentMonth) {
-                double[] dailyPercentageArray = getMonthDailyPercentageArray(previousMonth, isLeapYear);
-                totalPartialMonthCount = totalPartialMonthCount + dailyPercentageArray[previousMonthDayCount];
-                previousMonthDayCount = 0;
-            }
-            previousMonthDayCount++;
-        }
-        if (previousMonthDayCount > 0) {
-            double[] dailyPercentageArray = getMonthDailyPercentageArray(previousMonth, isLeapYear);
-            totalPartialMonthCount = totalPartialMonthCount + dailyPercentageArray[previousMonthDayCount];
+            return getNumberOfMonths(startDate, endDate);
         }
-        
-        return totalPartialMonthCount;
     }
-    
-    protected final double[] getMonthDailyPercentageArray(int month, boolean isLeapYear) {
-        switch(month) {
-            case Calendar.JANUARY:
-            case Calendar.MARCH:
-            case Calendar.MAY:
-            case Calendar.JULY:
-            case Calendar.AUGUST:
-            case Calendar.OCTOBER:
-            case Calendar.DECEMBER:
-                return THIRTY_ONE_DAY_COUNT;
-            case Calendar.FEBRUARY:
-                if (isLeapYear) {
-                    return TWENTY_NINE_DAY_COUNT;
-                } else {
-                    return TWENTY_EIGHT_DAY_COUNT;
-                }
-            case Calendar.APRIL:
-            case Calendar.JUNE:
-            case Calendar.SEPTEMBER:
-            case Calendar.NOVEMBER:
-                return THIRTY_DAY_COUNT;
-        }
-        throw new IllegalArgumentException("Illegal Month '" + month + "' passed in.");
-    }
-    
-    public static java.sql.Date rollMonths(Date startDate, int months ) {
-        return rollDate(startDate, Calendar.MONTH, months);
-    }
-    
-    public static java.sql.Date rollDays(Date startDate, int days ) {
-        return rollDate(startDate, Calendar.DATE, days);
+
+    protected double getNumberOfMonths(Date startDate, Date endDate) {
+    	DateTime start = new DateTime(startDate);
+    	DateTime end = new DateTime(endDate);
+    	double daysInMonthForEndDate = end.dayOfMonth().getMaximumValue();
+    	
+    	PeriodType periodType = PeriodType.standard().withWeeksRemoved();
+    	Period period = new Period(start, end, periodType);
+    	double monthsBetween = Months.monthsBetween(start, end).getMonths();
+    	double daysLeft = period.getDays();
```

Comment:
I don't know joda time API well, but what happens if a period is longer than a year? I believe you can manually do that right?